src/kontrol/kdist/cheatcodes.md

```diff
@@ -962,6 +968,60 @@ We use `#next[OP]` to identify OpCodes that represent function calls. If there i
       [priority(30)]
 ```
 
+```k
+    syntax KItem ::= "#setMockFunction" Account Account Bytes [symbol(foundry_setMockFunction)]
+ // ---------------------------------------------------------------------------------
+    rule <k> #setMockFunction MOCKADDRESS MOCKTARGET MOCKCALLDATA => .K ... </k>
+         <mockFunction>
+            <mockFunctionAddress> MOCKADDRESS </mockFunctionAddress>
+            <mockFunctionValues>  MOCKVALUES => MOCKVALUES [ MOCKCALLDATA <- MOCKTARGET ] </mockFunctionValues>
+         </mockFunction>
+
+   rule <k> #setMockFunction MOCKADDRESS MOCKTARGET MOCKCALLDATA => .K ... </k>
+         <mockFunctions>
+           ( .Bag
+            => <mockFunction>
+                  <mockFunctionAddress> MOCKADDRESS </mockFunctionAddress>
+                  <mockFunctionValues> .Map [ MOCKCALLDATA <- MOCKTARGET ] </mockFunctionValues>
+               </mockFunction>
+           )
+           ...
+         </mockFunctions>
+```
+
```

Comment:
Is it intentional that the definition of `mockFunction`-related rules is split into three `k` blocks separated by several empty lines? They could probably be combined into one?

Reply:
I've moved the `setMockFunction` section down after the `setMockCall` stuff in the utils section and combined `[cheatcode.call.mockFunction]` and `[foundry.set.mockFunction]` under their own header.

I'm not sure what you mean about the cleaning after a call is completed. It is intended to work similar to the `mockCall` cheatcode which I thought was supposed to mock all subsequent matching calls.

Reply:
Thanks! You're right about not having to clean `mockFunctions`, I didn't realise we want to mock subsequent matching calls.